### PR TITLE
IBP-3449 Implement BRAPI Get Studies

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java [hunk 1]
```diff
@@ -105,39 +107,73 @@
...
-		return new ResponseEntity<>(studiesList, HttpStatus.OK);
+		final boolean isSortOrderValid = "ASC".equals(sortOrder) || "DESC".equals(sortOrder) || StringUtils.isEmpty(sortOrder);
+		Preconditions.checkArgument(isSortOrderValid, "sortOrder should be either ASC or DESC");
+
+		final StudySearchFilter studySearchFilter = new StudySearchFilter();
+		final int finalPageNumber = currentPage == null ? BrapiPagedResult.DEFAULT_PAGE_NUMBER : currentPage + 1;
+		final int finalPageSize = pageSize == null ? BrapiPagedResult.DEFAULT_PAGE_SIZE : pageSize;
+		studySearchFilter.setCommonCropName(crop);
+		studySearchFilter.setStudyTypeDbId(studyTypeDbId);
+		studySearchFilter.setProgramDbId(programDbId);
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java]: `studyDbId `is not being set to filter

File: src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java [hunk 2]
```diff
@@ -105,39 +107,73 @@
...
 
-	@ApiOperation(value = "List of study summaries", notes = "Get a list of study summaries.")
-	// TODO implement
-	// @RequestMapping(value = "/{crop}/brapi/v1/studies", method = RequestMethod.GET)
+	@ApiOperation(value = "List of studies", notes = "Get a list of studies.")
+	@RequestMapping(value = "/{crop}/brapi/v1/studies", method = RequestMethod.GET)
 	@ResponseBody
-	public ResponseEntity<StudySummariesDto> listStudySummaries(@PathVariable final String crop,
+	public ResponseEntity<EntityListResponse<StudyDto>> listStudies(@PathVariable final String crop,
+		@ApiParam(value = "Common name for the crop associated with this study.") @RequestParam(value = "commonCropName", required = false)
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java]: Just a comment, maybe on the brapi spec.. It doesn't seem useful to include "commonCropName" as API parameter given we've already set the crop context in URL
cc: @nahuel-soldevilla

File: src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java [hunk 3]
```diff
@@ -105,39 +107,73 @@
...
-		final Pagination pagination = new Pagination();
-		final Metadata metadata = new Metadata().withPagination(pagination)
-			.withStatus(Collections.singletonList(Maps.newHashMap(ImmutableMap.of("message", "This call is not yet implemented."))));
-		final StudySummariesDto studiesList = new StudySummariesDto().setMetadata(metadata).setResult(results);
-
-		return new ResponseEntity<>(studiesList, HttpStatus.OK);
+		final boolean isSortOrderValid = "ASC".equals(sortOrder) || "DESC".equals(sortOrder) || StringUtils.isEmpty(sortOrder);
+		Preconditions.checkArgument(isSortOrderValid, "sortOrder should be either ASC or DESC");
+
+		final StudySearchFilter studySearchFilter = new StudySearchFilter();
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java]: Better create a `StudySearchFilter` constructor with the expected fields from parameter

File: src/main/java/org/ibp/api/java/impl/middleware/study/StudyServiceImpl.java [hunk 4]
```diff
@@ -383,4 +385,13 @@ public void updateStudy(final Study study) {
 		this.studyDataManager.updateStudyLockedStatus(studyId, study.isLocked());
 	}
 
+	@Override
+	public long countStudyDTOs(final StudySearchFilter studySearchFilter) {
```
Comment (darla-leafnode) [inline src/main/java/org/ibp/api/java/impl/middleware/study/StudyServiceImpl.java]: Rename to `countStudies` and `getStudies`

**Comment by darla-leafnode** [comment]
Also add an entry for this endpoint under calls.json file

File: src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java (hunk 2 above)
Comment (nahuel-soldevilla) [reply src/main/java/org/ibp/api/brapi/v1/study/StudyResourceBrapi.java]: We were following the [convention of accepting the parameter but then ignoring it](https://ibplatform.atlassian.net/browse/IBP-1841?focusedCommentId=35035&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-35035).

@darla-leafnode @abatac

**Pull request merged**